crypto/src/hash/pedersen/mod.rs

```diff
@@ -0,0 +1,102 @@
+use lambdaworks_math::{
+    cyclic_group::IsGroup,
+    elliptic_curve::short_weierstrass::{
+        curves::stark_curve::StarkCurve, point::ShortWeierstrassProjectivePoint,
+    },
+    field::{
+        element::FieldElement, fields::fft_friendly::stark_252_prime_field::Stark252PrimeField,
+    },
+};
+
+mod constants;
+mod parameters;
+use self::parameters::PedersenParameters;
+
+pub struct Pedersen {
+    params: PedersenParameters,
+}
+
+impl Default for Pedersen {
+    fn default() -> Self {
+        let pedersen_stark_default_params = PedersenParameters::default();
+        Self::new_with_params(pedersen_stark_default_params)
+    }
+}
+
+impl Pedersen {
+    pub fn new_with_params(params: PedersenParameters) -> Self {
+        Self { params }
+    }
+
+    // Taken from Jonathan Lei's starknet-rs
+    // https://github.com/xJonathanLEI/starknet-rs/blob/4ab2f36872435ce57b1d8f55856702a6a30f270a/starknet-crypto/src/pedersen_hash.rs
+    pub fn hash(
+        &self,
+        x: &FieldElement<Stark252PrimeField>,
+        y: &FieldElement<Stark252PrimeField>,
+    ) -> FieldElement<Stark252PrimeField> {
+        let add_points =
+            |acc: &mut ShortWeierstrassProjectivePoint<StarkCurve>,
+             bits: &[bool],
+             prep: &[ShortWeierstrassProjectivePoint<StarkCurve>]| {
+                bits.chunks(self.params.curve_const_bits)
+                    .enumerate()
+                    .for_each(|(i, v)| {
+                        let offset = bools_to_usize_le(v);
+                        if offset > 0 {
+                            // Table lookup at 'offset-1' in table for chunk 'i'
+                            *acc = acc.operate_with(&prep[i * self.params.table_size + offset - 1]);
+                        }
+                    });
+            };
```

Comment:
This seems like it could be a separate private function instead of defining it here. Is there any particular reason to do it?

Also, lets add some documentation about what this function does. Is it just adding elliptic curve points? Maybe the name could be better too, since it is not very clear what it does.

Reply:
Hey, thanks for the feedback.

The only reason was I tried to make as little changes as possible to the reference implementation. 

I now defined it as a private function as you suggested, and renamed it into "lookup_and_accumulate", as well as putting some explanations. I'm not sure if this is a great name but it sounds more explanatory.